packages/generator/src/parser/parser-types-common.ts

```diff
@@ -1,109 +1,66 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 
 import { PathLike } from 'fs';
+import { ODataVersion } from '@sap-cloud-sdk/util';
 
-export interface ParsedServiceMetadata {
-  edmx: EdmxMetadata;
-  swagger?: SwaggerMetadata;
-}
-
-export interface EdmxMetadata {
+export interface EdmxMetadataBase {
   path: PathLike;
+  oDataVersion: ODataVersion;
   fileName: string;
   namespace: string;
-  entityTypes: EdmxEntityType[];
-  entitySets: EdmxEntitySet[];
-  associationSets: EdmxAssociationSet[];
-  associations: EdmxAssociation[];
-  functionImports: EdmxFunctionImport[];
-  complexTypes: EdmxComplexType[];
   selfLink?: string;
 }
 
-export interface EdmxAssociation {
-  Name: string;
-  'sap:content-version': string;
-  End: EdmxAssociationEnd[];
-}
-
-interface EdmxAssociationEnd {
-  Type: string;
-  Multiplicity: string;
-  Role: string;
+export interface EdmxMetadataBaseExtended extends EdmxMetadataBase {
+  entitySets: EdmxEntitySetBase[];
+  entityTypes: EdmxEntityTypeBase[];
+  complexTypes: EdmxComplexTypeBase[];
+  functionImports: EdmxFunctionImportBase[];
 }
 
-export interface EdmxAssociationSet {
-  Name: string;
-  Association: string;
-  'sap:creatable': string;
-  'sap:updatable': string;
-  'sap:deletable': string;
+export interface EdmxEntityTypeBase extends EdmxNamed {
+  Key: EdmxKey;
+  Property: EdmxProperty[];
+  NavigationProperty: any[];
   'sap:content-version': string;
-  End: EdmxAssociationSetEnd[];
-}
-
-interface EdmxAssociationSetEnd {
-  Role: string;
-  EntitySet: string;
+  'sap:label'?: string;
 }
 
-export interface EdmxFunctionImport {
-  Name: string;
+export interface EdmxFunctionImportBase extends EdmxNamed {
   ReturnType: string;
-  'm:HttpMethod': string;
   'sap:action-for': string;
   Parameter: EdmxParameter[];
   EntitySet?: string;
 }
 
-export interface EdmxParameter extends EdmxDocumented {
-  Name: string;
+export interface EdmxParameter extends EdmxDocumented, EdmxNamed {
   Type: string;
   Nullable: string;
 }
 
-export interface EdmxComplexType {
-  Name: string;
+export interface EdmxComplexTypeBase extends EdmxNamed {
   Property: EdmxProperty[];
 }
 
-export interface EdmxEntitySet {
+export interface EdmxEntitySetBase extends EdmxNamed {
   EntityType: string;
-  Name: string;
   'sap:content-version': string;
```

Comment:
I find these kind of strings e.g. `sap:content-version` in multiple places in the code. Does it make sense to replace them via a key collection or strings?

Reply:
What do you mean by key collection? I think those are read when parsing at some point, but maybe when we do your proposed refactoring this can also be considered? I would not want to add this to this PR ;)

Reply:
ok fair, I think about having a interface or so with these keys and use it then in this type to minimize the occurrence of any explicit string.

Reply:
We can probably lose them alltogether in the refactoring.